swayfmt/src/utils/map/newline.rs

```diff
@@ -45,59 +39,50 @@ fn is_new_line_in_rope(rope: &Rope, index: usize) -> bool {
     true
 }
 
-/// Checks if there is a new line (Operating system specific) next in the iter of characters
-#[inline]
-fn is_new_line_next_in_iter(
-    input_iter: &mut Peekable<Enumerate<Chars<'_>>>,
-    new_line: &[char],
-) -> bool {
-    let total = new_line.len();
-    for (p, char_new_line) in new_line.iter().enumerate() {
-        if input_iter.peek().map(|x| x.1) == Some(*char_new_line) {
-            if total != p + 1 {
-                input_iter.next();
-            }
-        } else {
-            return false;
-        }
-    }
-
-    true
-}
-
 /// Search for newline sequences in the unformatted code and collect ByteSpan -> NewlineSequence for the input source
 fn newline_map_from_src(unformatted_input: &str) -> Result<NewlineMap, FormatterError> {
     let mut newline_map = BTreeMap::new();
     // Iterate over the unformatted source code to find NewlineSequences
-    let mut input_iter = unformatted_input.chars().enumerate().peekable();
+    let mut input_iter = unformatted_input.chars().peekable();
     let mut current_sequence_length = 0;
     let mut in_sequence = false;
     let mut sequence_start = 0;
-    let os_new_line = NEW_LINE.chars().collect::<Vec<_>>();
-    while let Some((char_index, char)) = input_iter.next() {
-        let next_char = input_iter.peek().map(|input| input.1);
-        if matches!(char, '/' | ';' | '}')
-            && is_new_line_next_in_iter(&mut input_iter, &os_new_line)
-        {
+    let mut bytes_offset = 0;
+    while let Some(char) = input_iter.next() {
```

Comment:
Let's move this to a separate PR so it's clear what bug this is fixing. It will be easier to review separately.